### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `env()`

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#6635]
### What steps are needed to reproduce the bug?

```css
a {
  margin: env(safe-area-inset-bottom);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "declaration-property-value-no-unknown": true,
}
```

### How did you run Stylelint?

CLI and VSCode

### Which version of Stylelint are you using?

15.0.0

### What did you expect to happen?

I expected `env()` to be allowed in the same way as `var()`

### What actually happened?

```txt
 ⚠  Unexpected unknown value "env(" for property "padding-bottom"  declaration-property-value-no-unknown
```

**Comment by Mouvedia** [comment]
That was to be expected. I'd say since `env`, `clamp` and the other unsupported css functions are included in [css-functions-list](https://github.com/niksy/css-functions-list/blob/master/index.json) which is already used by the `function-no-unknown` rule, we should reuse it until csstree gets patched.
[The 2 PR](https://github.com/stylelint/stylelint/pull/6511#issuecomment-1416753231) that I sent 1 week ago will probably remain unanswered for a long time, so adding another wouldn't help; let's fix it on our side for now.

**Comment by jeddy3** [comment]
@romainmenke Thanks for the report and for using the template.

Let's:

- patch upstream (either in [CSSTree](https://github.com/csstree/csstree/blob/master/data/patch.json) or [mdn-data](https://github.com/mdn/data/), which ever is appropriate)
- workaround it here in Stylelint (as @Mouvedia [mentioned](https://github.com/stylelint/stylelint/issues/6635#issuecomment-1425766407))

For the latter, we can take the same approach as `clamp/min/max`, i.e. returning early if an `env` node is present.

[Spec for `env()` for reference](https://w3c.github.io/csswg-drafts/css-env/#env-function).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

---

@Mouvedia Thank you for triaging the issue. [Our conventions for triaging them consistently](https://stylelint.io/maintainer-guide/issues) are in the maintainer guide.

**Comment by romainmenke** [comment]
- `mdn-data` already includes `env()` in the same way as it does `var()`
- [opened a PR to passthrough `env()` in the css tree lexer](https://github.com/csstree/csstree/pull/245)
